marshal/consumer.go

```diff
@@ -235,12 +235,47 @@ func (c *Consumer) defaultTopicPartitions() int {
 	return 0
 }
 
+// claimTerminated is called by a claim when they've terminated. This is used so we can
+// ensure topic claim semantics are adhered to. In topic claim mode this will be called
+// by every claim during a release.
+func (c *Consumer) claimTerminated(cl *claim, released bool) {
+	// For now, we don't care except in the topic claim mode
+	if !c.options.ClaimEntireTopic {
+		return
+	}
+
+	latestClaims, claimsErr := c.GetCurrentTopicClaims()
+	if claimsErr != nil {
+		// This only impacts update topic claims later
+		log.Errorf("claimTerminated: failed getting current claims: %s", claimsErr)
+	}
+	delete(latestClaims, cl.topic)
```

Comment:
Alright. Maybe in the future let's avoid that kind of design -- or at least strongly document it somewhere but I don't see any comments to that fact?